### ISSUE TITLE
[BUG] Unable to play .m3u8 file

### ISSUE THREAD

**Issue by masoudk2000** [jhomlala/betterplayer#812]
I have a .m3u8 file that contains information about a video but I am unable to play it using the BetterPlayer Flutter package. This is my sample code

var dataSource = BetterPlayerDataSource(
    BetterPlayerDataSourceType.file,
    "/var/mobile/Containers/Data/Application/B0391873-25D4-471A-AB13-05D5FF8701F9/Documents/content.publicId/test.m3u8",
  );
  
These are the contents of the .m3u8 file:

#EXTM3U
#EXT-X-VERSION:3
#EXT-X-ALLOW-CACHE:YES
#EXT-X-TARGETDURATION:37
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-KEY:METHOD=AES-128, URI="https://PATH_TO_ENCRYPTION_KEY",IV=0x413873f3924c752d99c7083c0c99931c
#EXTINF:36.440000, https://example.com/test_480p_0000.ts
#EXTINF:28.040000, https://example.com/test_480p_0001.ts
#EXTINF:28.360000, https://example.com/test_480p_0002.ts
#EXTINF:29.920000, https://example.com/test_480p_0003.ts
#EXTINF:36.920000, https://example.com/test_480p_0004.ts
#EXTINF:21.040000, https://example.com/test_480p_0005.ts
#EXT-X-ENDLIST

But I keep getting this error:

  GetDataFromUrl failed: Invalid argument(s): No host specified in URI /var/mobile/Containers/Data/Application/FE19C886-73B8-49C0-A770-6BAEAD354947/Documents/content.publicId/test.m3u8
  
  Failed to load video: Error Domain=CoreMediaErrorDomain Code=-12865
  "(null)" [VERBOSE-2:ui_dart_state.cc(209)] Unhandled Exception:
  PlatformException(VideoError, Failed to load video: The operation
  couldn’t be completed. (CoreMediaErrorDomain error -12865.), null,
  null)

**Better Player version**
- v0.0.79:

**Smartphone (please complete the following information):**
all IOS devices

**Comment by si-muntazir-rizvi** [comment]
Any luck on this error because I'm facing the same while playing .m3u8 video on iOS.

**Comment by FCE-tc** [comment]
BetterPlayerDataSourceType.file should be BetterPlayerDataSourceType.network,

> I have a .m3u8 file that contains information about a video but I am unable to play it using the BetterPlayer Flutter package. This is my sample code
> 
> This code works well in Android but not in ios
> 
> var dataSource = BetterPlayerDataSource( BetterPlayerDataSourceType.file, "/var/mobile/Containers/Data/Application/B0391873-25D4-471A-AB13-05D5FF8701F9/Documents/content.publicId/test.m3u8", );
> 
> These are the contents of the .m3u8 file:
> 
> #EXTM3U #EXT-X-VERSION:3 #EXT-X-ALLOW-CACHE:YES #EXT-X-TARGETDURATION:37 #EXT-X-MEDIA-SEQUENCE:0 #EXT-X-KEY:METHOD=AES-128, URI="https://PATH_TO_ENCRYPTION_KEY" #EXTINF:36.440000, https://example.com/test_480p_0000.ts #EXTINF:28.040000, https://example.com/test_480p_0001.ts #EXTINF:28.360000, https://example.com/test_480p_0002.ts #EXTINF:29.920000, https://example.com/test_480p_0003.ts #EXTINF:36.920000, https://example.com/test_480p_0004.ts #EXTINF:21.040000, https://example.com/test_480p_0005.ts #EXT-X-ENDLIST
> 
> But I keep getting this error:
> 
> GetDataFromUrl failed: Invalid argument(s): No host specified in URI /var/mobile/Containers/Data/Application/FE19C886-73B8-49C0-A770-6BAEAD354947/Documents/content.publicId/test.m3u8
> 
> Failed to load video: Error Domain=CoreMediaErrorDomain Code=-12865 "(null)" [VERBOSE-2:ui_dart_state.cc(209)] Unhandled Exception: PlatformException(VideoError, Failed to load video: The operation couldn’t be completed. (CoreMediaErrorDomain error -12865.), null, null)
> 
> **Better Player version**
> 
> * v0.0.79:
> 
> **Smartphone (please complete the following information):** all IOS devices

**Comment by jhomlala** [comment]
Do you have permission to access this file? `/var/mobile/Containers/Data/Application/B0391873-25D4-471A-AB13-05D5FF8701F9/Documents/content.publicId/test.m3u8` 
Can you try to access this file with dart's `File` API? Try to check if you're able to read this file from dart code.

**Comment by masoudk2000** [comment]
@jhomlala  yes,I have access to this file and i can read and write to this file

**Comment by DivineTechs** [comment]
same issue we have. 

not play m3u8 video

**Comment by katerynapeikova** [comment]
Also have same issue.

**Comment by BelongsH** [comment]
Also have same issue. when play local m3u8 file in ios . android device is normal~

**Comment by pedraohenrique** [comment]
The same here...

I have access to file with dart's File API, but I cant play the video.

I've also tried to put 'file://' at the start of filename, but it still doesn't works.

**Comment by usilitel** [comment]
Any progress? I have the same problem with LOCAL m3u8 files. Remote m3u8 files are working.
